eo-maven-plugin/src/test/java/org/eolang/maven/optimization/OptCachedTest.java

```diff
@@ -50,73 +49,92 @@ final class OptCachedTest {
-        OptCachedTest.save(tmp, program);
+    void returnsFromCacheButLastModifiedTimesDifferent(
+        @TempDir final Path cache,
+        @TempDir final Path dir
+    )
+        throws Exception {
+        final XML xml = OptCachedTest.program();
+        final Path program = OptCachedTest.save(dir, xml);
+        OptCachedTest.save(cache, xml, 2000);
         MatcherAssert.assertThat(
-            "We expected that the not immediately saved program will be returned from the cache.",
+            "We expected that the program will be returned from the cache.",
             new OptCached(
                 path -> {
                     throw new IllegalStateException("This code shouldn't be executed");
                 },
-                tmp
-            ).apply(program),
-            Matchers.equalTo(program)
+                cache,
+                program
+            ).apply(xml),
+            Matchers.equalTo(xml)
         );
     }
 
-    @Disabled
     @Test
-    void returnsFromCacheCorrectProgram(@TempDir final Path tmp)
-        throws IOException {
-        final XML prev = OptCachedTest.program(ZonedDateTime.now(), "first program");
-        OptCachedTest.save(tmp, prev);
-        final XML current = OptCachedTest.program(ZonedDateTime.now(), "second program");
+    void returnsFromCacheCorrectProgram(@TempDir final Path cache, @TempDir final Path dir)
+        throws Exception {
+        final XML first = OptCachedTest.program("first program");
+        final XML second = OptCachedTest.program("second program");
+        OptCachedTest.save(cache, first, -2000);
+        final Path current = OptCachedTest.save(dir, second);
         MatcherAssert.assertThat(
             "Expecting current program to be compiled, but prev program was returned from cache.",
-            new OptCached(
-                path -> current,
-                tmp
-            ).apply(current),
-            Matchers.equalTo(current)
+            new OptCached(path -> second, cache, current)
+                .apply(second),
+            Matchers.equalTo(second)
         );
     }
 
     @Test
-    void optimizesIfXmlIsAbsentInCache(@TempDir final Path tmp) {
-        final XML program = OptCachedTest.program();
-        final Path cache = tmp.resolve("cache");
+    void optimizesIfXmlIsAbsentInCache(@TempDir final Path cache, @TempDir final Path dir)
+        throws Exception {
+        final Path program = OptCachedTest.save(dir, OptCachedTest.program());
         MatcherAssert.assertThat(
             "We expect that the program will be created and returned as is (same instance)",
-            new OptCached(path -> program, cache).apply(program),
-            Matchers.sameInstance(program)
+            new OptCached(
+                path -> OptCachedTest.program(), cache, program
+            )
+                .apply(OptCachedTest.program()),
+            Matchers.equalTo(OptCachedTest.program())
```

Comment:
@Yanich96 Can we just create a sing variable `prog = OptCachedTest.program()` and use it in all four places?